generators/chipyard/src/main/scala/TestHarness.scala

```diff
@@ -26,5 +24,15 @@ class TestHarness(implicit val p: Parameters) extends Module {
 
   val dut = p(BuildTop)(p)
   io.success := false.B
-  p(IOBinders).values.map(fn => fn(clock, reset.asBool, io.success, dut))
+  // resetOverride can be overridden via a harnessFunction
+  val resetOverride = Wire(Bool())
+  resetOverride := reset
+  dut.harnessFunctions.foreach(_(this))
+
+  // Aliases for clock, reset, and success
+  def c  = clock
+  def r  = reset.asBool
```

Comment:
Do functions ever need to disambiguate between reset and the resetOverride? AFAICT resetOverride _is_ the reset, with the added bonus it can be assigned to.  I feel like something like this will present
and implicit reset that that is more intuitive to the user: 

```val resetOverride = WireDefault(Bool())
withClockAndReset(clock, resetOverride) {
     dut.harnessFunctions.foreach(_(this))
   }
}
```
and then remove the reset alias "r" as it currently exists.

Reply:
> > Also can we inline IceNet into TestChipIp?
> 
> You mean merge icenet and testchipip into one submodule? Why?

No opinions on the merge, but it does feel a bit out of scope for this PR.

Reply:
Yeah, I like that. Good suggestion.

Reply:
We might need to keep the `r` alias though, to support this kind of thing (from the `WithSimDebug` binder):
```
th.ro := th.r | debugPortOpt.map { debug => AsyncResetReg(debug.ndreset).asBool }.getOrElse(false.B)
```

Open to suggestions about giving it a better name, though.

Reply:
I went back and forth on this a bit. I think I've actually decided to scrap all of the aliases and make people use the full names- it adds a lot of clarity IMO. I also think we don't want the `withClockAndReset`, because generally I think the test harness widgets will want to be on the `TestHarness` reset, not `dutReset` (I renamed `resetOverride` to `dutReset`, which I think is clearer). If we find that in the future we need some advanced reset sequencing we can rethink this.

Reply:
yeah that's much clearer.